ppl/zqd/auth.go

```diff
@@ -84,20 +120,57 @@ func newAuthenticator(ctx context.Context, logger *zap.Logger, registerer promet
 		},
 		SigningMethod: jwt.SigningMethodRS256,
 	})
-	return func(next http.Handler) http.Handler {
-		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			if err := checker.CheckJWT(w, r); err != nil {
-				// response sent by jwtmiddleware.Options.ErrorHandler
-				return
-			}
-			next.ServeHTTP(w, r)
-		})
+	return &Auth0Authenticator{
+		checker: checker,
+		methodResponse: api.AuthMethodResponse{
+			Kind: api.AuthMethodAuth0,
+			Auth0: &api.AuthMethodAuth0Details{
+				Domain:   config.Domain,
+				ClientID: config.ClientID,
+			},
+		},
 	}, nil
 }
 
+func (a *Auth0Authenticator) Middleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if err := a.checker.CheckJWT(w, r); err != nil {
+			// response sent by jwtmiddleware.Options.ErrorHandler
+			return
+		}
+		next.ServeHTTP(w, r)
+	})
+}
+
+func (a *Auth0Authenticator) MethodResponse() api.AuthMethodResponse {
+	return a.methodResponse
+}
+
+func verifyAPIAudience(claims jwt.MapClaims) bool {
+	// Audience claim may either be a string, or a slice of interfaces that are
+	// strings.
+	// https://auth0.com/docs/tokens/access-tokens/get-access-tokens#multiple-audiences
+	if str, ok := claims["aud"].(string); ok {
```

Comment:
Should `aud` be a `const` like TenetIDClaim and UserIDClaim?

Reply:
It's standard (ie, well known) claim for a JWT (like "GET" as an HTTP verb), unlike our custom claims for user & tenant id.